frame/uniques/src/impl_nonfungibles.rs

```diff
@@ -106,3 +106,25 @@ impl<T: Config<I>, I: 'static> Transfer<T::AccountId> for Pallet<T, I> {
 		Self::do_transfer(class.clone(), instance.clone(), destination.clone(), |_, _| Ok(()))
 	}
 }
+
+impl<T: Config<I>, I: 'static> InspectEnumerable<T::AccountId> for Pallet<T, I> {
+	/// Returns the asset classes in existence.
+	fn classes() -> Vec<Self::ClassId> {
+		ClassMetadataOf::<T, I>::iter().map(|(k, _v)| k).collect()
+	}
+
+	/// Returns the instances of an asset `class` in existence.
+	fn instances(class: &Self::ClassId) -> Vec<Self::InstanceId> {
+		InstanceMetadataOf::<T, I>::iter_prefix(class).map(|(i, _v)| i).collect()
+	}
```

Comment:
**[Optional Suggestion]**
I think we must alert the users that this operation is dangerous, once an Account may have million instances.
```suggestion
	/// Returns the instances of an asset `class` in existence.
	/// WARNING: This may be a heavy operation. Do not use when execution time is limited.
	fn instances(class: &Self::ClassId) -> Vec<Self::InstanceId> {
		InstanceMetadataOf::<T, I>::iter_prefix(class).map(|(i, _v)| i).collect()
	}
```

Reply:
**[Optional Suggestion]**
Possible it can be done in another PR, but this trait is not very useful right now once all on-chain operations are time limited, I see three ways to solve this:

1 - Pagination: But AFAIK StorageMap and StorageNMap doesn't support pagination, maybe a new storage class will be required.

2 - Return a Iterable instead of a Vec, so the caller have control over how many instances he will iterate.
```rust
/// Returns the instances of an asset `class` in existence.
fn instances(class: &Self::ClassId) -> Iterator<Self::InstanceId> {
	InstanceMetadataOf::<T, I>::iter_prefix(class).map(|(i, _v)| i)
}
```

3 - Use **BoundedVec** combined with **Iterable::take()** (obs: I did not test this code)
```rust
/// Returns the instances of an asset `class` in existence.
fn instances<T: Get<u32>(class: &Self::ClassId) -> BoundedVec<Self::InstanceId, T> {
	BoundedVec::<Self::InstanceId, T>::try_from(
		InstanceMetadataOf::<T, I>::iter_prefix(class).map(|(i, _v)| i).take(T::get()).collect()
	).unwrap()
}
```

Reply:
yeah these functions should only be used in environments like tests or some off-chain service.

There will be no way to safely iterate over things like this in the runtime, and probably it would not be sensible to write any code which approaches tokens like this.

Reply:
I think there's situations where we may want to iterate over them on-chain, for example the pagination can be useful in some migration that uses the `on_initialize` hook to lazy migrate some data, or a smart-contracts that needs to list the tokens of a given account, etc.

Reply:
@Lohann done!